libs/barista-components/table/src/table-order-data-source.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DataSource } from '@angular/cdk/table';
+import { BehaviorSubject, Observable, Subscription, of } from 'rxjs';
+import { map, withLatestFrom } from 'rxjs/operators';
+
+import { DtTable } from './table';
+import { DtOrder, DtOrderChangeEvent, DtOrderReorderEvent } from './order';
+import { moveItemInArray } from '@angular/cdk/drag-drop';
+
+export class DtTableOrderDataSource<T> extends DataSource<T> {
+  /** Stream that emits when a new data array is set on the data source. */
+  private _data: BehaviorSubject<T[]>;
```

Comment:
Why do we need to reimplement this if we extend from the datasource which already holds the exact same property?